Common/Securities/NonShortableCashAccountMarginModel.cs

```diff
@@ -0,0 +1,158 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Orders;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="ISecurityMarginModel"/> that requires cash of the correct
+    /// currency to exist in the portfolio to make trades. This also only applies to securities where shorting
+    /// is not an option. IOW, holdings.Quantity >= 0 at all times.
+    /// </summary>
+    public class NonShortableCashAccountMarginModel : ISecurityMarginModel
```

Comment:
Perhaps this should just be `CashAccountMarginModel` -- Do any cash accounts allow you to naked short an asset?